windows/hid.c

```diff
@@ -302,46 +302,74 @@ int HID_API_EXPORT hid_exit(void)
 	return 0;
 }
 
-static void hid_internal_get_ble_info(struct hid_device_info* dev, DEVINST dev_node)
+static wchar_t* hid_internal_get_devnode_property(DEVINST dev_node, const DEVPROPKEY* property_key, DEVPROPTYPE expected_property_type)
 {
```

Comment:
```suggestion
static wchar_t* hid_internal_get_wchar_t_devnode_property(DEVINST dev_node, const DEVPROPKEY* property_key, DEVPROPTYPE expected_property_type)
{
	if (expected_property_type != DEVPROP_TYPE_STRING || expected_property_type != DEVPROP_TYPE_STRING_LIST)
		return NULL;
```

Reply:
there is `property_type != expected_property_type` check in that method already.
what if I just change `wchar_t*` to `PVOID` return type?

Reply:
That's one way of doing it.
But you'd have to cast the return value to `wchar_t *`  everywhere you use it.

maybe have both: generic `hid_internal_get_devnode_property` that returns `PVOID` _and_ `hid_internal_get_wchar_t_devnode_property` that does the check suggested above and returns `wchar_t`?